From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Thu, 10 Oct 2024 14:54:10 +0200
Subject: privacy: Handle setting combo widgets to unknown gsetting values

The idle-delay and lock-delay settings are read from gsettings, which
the user might have set to values not present in the enum model.

Ideally we would add the unknown value to the model, like equivalent
idle-delay setting from the power panel. But these widgets are set-up
with AdwEnumListModel's and changing that would require a refactoring.

Forwarded: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/2873
---
 panels/privacy/cc-screen-page.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/panels/privacy/cc-screen-page.c b/panels/privacy/cc-screen-page.c
index 2527da8..8f28679 100644
--- a/panels/privacy/cc-screen-page.c
+++ b/panels/privacy/cc-screen-page.c
@@ -99,6 +99,9 @@ on_lock_combo_changed_cb (AdwComboRow   *combo_row,
   AdwEnumListItem *item;
   CcScreenPageLockAfter delay;
 
+  if (adw_combo_row_get_selected (combo_row) == GTK_INVALID_LIST_POSITION)
+    return;
+
   item = ADW_ENUM_LIST_ITEM (adw_combo_row_get_selected_item (combo_row));
   delay = adw_enum_list_item_get_value (item);
 
@@ -181,6 +184,9 @@ on_blank_screen_delay_changed_cb (AdwComboRow   *combo_row,
   AdwEnumListItem *item;
   CcScreenPageBlankScreenDelay delay;
 
+  if (adw_combo_row_get_selected (combo_row) == GTK_INVALID_LIST_POSITION)
+    return;
+
   item = ADW_ENUM_LIST_ITEM (adw_combo_row_get_selected_item (combo_row));
   delay = adw_enum_list_item_get_value (item);
 
